frontend/public/components/instantiate-template.tsx

```diff
@@ -129,55 +129,30 @@ const stateToProps = (state: RootState) => ({
   models: state.k8s.getIn(['RESOURCES', 'models']),
 });
 
-class TemplateForm_ extends React.Component<
-  TemplateFormProps & WithTranslation,
-  TemplateFormState
-> {
-  constructor(props: TemplateFormProps & WithTranslation) {
-    super(props);
+const TemplateForm_: React.FC<TemplateFormProps> = (props) => {
+  const { preselectedNamespace: ns = '', activePerspective, perspectiveExtensions, obj } = props;
 
-    const { preselectedNamespace: namespace = '' } = this.props;
-    const parameters = this.getParameterValues(props);
-    this.state = {
-      namespace,
-      parameters,
-      inProgress: false,
-      error: '',
-    };
-  }
+  const [namespace, setNamespace] = React.useState(ns);
+  const [parameters, setParameters] = React.useState([]);
+  const [inProgress, setInProgress] = React.useState(false);
+  const [error, setError] = React.useState('');
 
-  componentDidUpdate(prevProps: TemplateFormProps & WithTranslation) {
-    if (this.props.obj.data?.parameters !== prevProps.obj.data?.parameters) {
-      const parameters = this.getParameterValues(this.props);
-      this.setState({ parameters });
-    }
-  }
+  const { t } = useTranslation();
 
-  getParameterValues = (props: TemplateFormProps & WithTranslation) => {
-    const templateParameters: TemplateParameter[] = props.obj.data.parameters || [];
-    return templateParameters.reduce((acc, { name, value }: TemplateParameter) => {
+  React.useEffect(() => {
+    const object = (obj.data.parameters || []).reduce((acc, { name, value }) => {
       acc[name] = value;
       return acc;
     }, {});
-  };
-
-  onNamespaceChange = (namespace: string) => {
-    this.setState({ namespace });
-  };
+    setParameters(object);
+  }, [obj]);
 
-  onParameterChanged: React.ReactEventHandler<HTMLInputElement> = (event) => {
+  const onParameterChanged: React.ReactEventHandler<HTMLInputElement> = (event) => {
```

Comment:
We could use `React.useCallback` here to memoize the handler function, as well as any other functions below that are used in JSX as component props. Can be done later as an improvement.